### ISSUE TITLE
Howto more 'undetectable'

### ISSUE THREAD

**Issue by whowhere** [wayneaswilliams/msfvenom_custom_encoding#3]
Hi Wayne,

I like your scripts also the old commands script :)
I'm just trying to learn more and understand more, but I don't understand what makes the payloads more 'undetectable' as all payloads nomatter what encrypter I'm using is detected.

**Comment by wayneaswilliams** [comment]
Hey,
The best way to explain is to take a picture of yourself in the mirror.
Then take a picture with a scarf around your face in the mirror and notice
the difference. Antivirus companies usually employ signature based
detection (with the exception of Symantec's SONAR and things of that sort),
which means if it looks like a virus they know about, it gets detected.
When generating a payload with MSFVenom, using -f exe , by default it uses
a payload template that just replaces {PAYLOAD} with the shellcode
generated. This will be easily detected by signature based Antivirus.
That's why this script creates a template on the fly. Although I haven't
worked with this one in a long time, and have newer revisions that I
haven't released, the idea is still the same.

W. Alexander Williams, OSCP
wayneaswilliams@gmail.com
1 (919) 949-8986
Resume <http://wayneaswilliams.com/resume>

**Comment by dnsBlah** [comment]
Hey Wayne,

Thanks for your reponse! I do get the "idea" of 'masqurading', but I'm not sure what part actually helps the most
- some random application code, like... your loop with echo's
-- I'd do it a bit different maybe generating an image or whatever
- or is it the encoding part from venom, shikata etc ?
-- I used multiples and others and combine them, but the result stayed the same.

Stupid smartscreen keeps kicking in, kasparsky doesn't (anymore)

**Comment by whowhere** [comment]
Hey Wayne,

Thanks for your reponse! I do get the "idea" of 'masqurading', but I'm not sure what part actually helps the most

-some random application code, like... your loop with echo's
    -- I'd do it a bit different maybe generating an image or whatever
-or is it the encoding part from venom, shikata etc ?
    -- I used multiples and others and combine them, but the result stayed the same.

Stupid smartscreen keeps kicking in, kasparsky doesn't (anymore)

**Comment by wayneaswilliams** [comment]
Encoding really does nothing anymore but remove bad characters. I used that
encoding procedure to make it bigger which can throw off signatures. Try
adding a "sleep" timer to your C code. However, make sure its not literally
the sleep function. Make the system work for a bit before executing your
code. I remember one revision I did was pushed a bunch of asterisks in
command line for about 10000 loops which makes the computer run that code
for about a minute before executing the shell code.

W. Alexander Williams, OSCP
wayneaswilliams@gmail.com
1 (919) 949-8986
Resume <http://wayneaswilliams.com/resume>
